src/cha.c

```diff
@@ -474,6 +466,14 @@ static void cha_loop_transfer_callback(struct libusb_transfer* transfer) {
 				};
 			}
 
+			if (!(done || loop->break_loop) && (ret = libusb_submit_transfer(transfer)) < 0) {
+				if (ret != LIBUSB_ERROR_INTERRUPTED) {
+					cha->error_str = libusb_error_name(ret);
+				}
+
+				loop->break_loop = 1;
+			}
+
```

Comment:
The same explanation applies as in the write case, except now the data flow is opposite. The driver can write the data (and it is trivial to experience on Windows on my computer) while we haven't yet parsed the data that was in the buffer.

Reply:
What do you think? I would like to introduce second `libusb_transfer` object here in order not to wait for parser done before firing next data request to the hardware.

Reply:
I think it might be worthwhile to do so alongside some refactor (that is outside the scope of bug fix).

If you look at current extcap implementation it calls this function with `count` equal 1 - and thus the second transfer here won't help. This is because Wireshark lets extcap know that it should stop by closing the pipe. If there was mechanism to let the user callback to notify library whether it should continue or not (possibly instead of the count) then it would make sense in my opinion to have two transfers here.

Reply:
Actually, there is the mechanism. `ov_capture_dispatch` should properly handle SIGTERM and co. This is how pcap modules work: https://github.com/matwey/libpcap/commit/48341c7dce3ad3ea0ec369d4e2d1e6454d23f352

Reply:
I am probably missing something, but I can't see anything related to SIGTERM in the linked commit. Neither I see anything in the library except the sample that has SIGTERM handler.

Am I unerstanding it correctly to simply `raise(SIGTERM);` inside the callback?

Reply:
In Linux, signals are received when application makes a syscall. Then the syscall is returned with special error code and registered handler is executed before that.

So, if `LIBUSB_ERROR_INTERRUPTED` is returned from `libusb_handle_events_completed`, then it means that we are received a signal, so the function is terminated with special error code -2 in order to show the user that the loop has been interrupted by a signal.

Reply:
I think that I am going to merge this PR and we will recover the performance later.